readthedocs/proxito/middleware.py

```diff
@@ -114,3 +115,20 @@ def process_request(self, request):  # noqa
         request.host_project_slug = request.slug = ret
 
         return None
+
+    def process_response(self, request, response):  # noqa
+        """Set the Strict-Transport-Security (HSTS) header for a custom domain if max-age>0."""
```

Comment:
Just so I'm clear, you're envisioning that we should set the HSTS header in proxito (if appropriate) exclusively in middleware whether it's for a custom domain or for the public domain (`*.readthedocs.io`)?

You're still planning on relying on nginx setting the HSTS header for the dashboard/production domain (readthedocs.org), correct? Because we want this header set for some domains and not others, we can't rely on the usual method of Django's [SecurityMiddleware](https://docs.djangoproject.com/en/2.2/ref/middleware/#module-django.middleware.security).

Reply:
Yep, just for the requests through proxito on the PUBLIC_DOMAIN and probably EXTERNAL_DOMAIN (PR builders) as well.